### DISCUSSION TITLE
Add Redis protocol support

### DISCUSSION THREAD

**Discussion by vinerich** [ory/kratos#1721] [Ideas]
This ticket is created to collect feedback for the integration of the `redis` database protocol.

As currently seen in the [configuration](https://www.ory.sh/kratos/docs/next/reference/configuration/) supported protocols for the database storage are
- postrgres
- mysql
- cockroach
- sqlite

With the support of the `redis` protocol many more databases such as
- redis
- keyDB
- titan
- ...

will be supported which may widen the target audience or just the integration flexibility into existing infrastructure.

If you would like to see `redis` natively integrated with kratos please add a 👍 or a ❤️ to this post rather than writing "+1" in the comments.

To give credits for the last sentence I took it from [here](https://github.com/ory/oathkeeper/issues/470)
For any devs feel free to close the issue or request edits if this is not the right place or the right scope for the topic!

Edit: Fixed link formatting

**Comment by aeneasr** [comment]
Hello, we are planning to add support for Redis for certain latency-sensitive operations like looking up sessions, but have no plans and probably never will add it as a primary storage option.

**Reply by vinerich** [reply]
Hey! Thanks for moving the issue to continue in a more appropriate place!

Integrating Redis-Protocol for certain latency-sensitive operations like looking up sessions would be sufficient in my opinion. I could edit the conversation to reflect this use case rather than using it as a primary storage option.

However I sadly didn't find anything regarding Redis (or databases in general) in the [Milestones and Roadmap](https://www.ory.sh/kratos/docs/next/milestones/). Is this feature already planned (timewise) or more in planning stage?

**Reply by aeneasr** [reply]
It's currently in the back of my head - we have not tracked it individually. We will definitely address this as soon as we hit some scalability issues at console.ory.sh - so it's kinda of a guarantee that we will add it at some point :)

**Reply by vinerich** [reply]
Thanks for answering! I'll wait patiently but a big 👍🏾 from my side :)

**Reply by aeneasr** [reply]
np :)
